ToNote/ViewModels/MainViewModel.cs

```diff
@@ -18,6 +21,14 @@ public MainViewModel()
             {
                 using (var reader = new StreamReader(file))
                     Notes.Add(JsonConvert.DeserializeObject<Note>(reader.ReadToEnd()));
+
+                SelectedTags.CollectionChanged += (s, e) =>
+                {
+                    if (SelectedTags.Any())
```

Comment:
You don't need this check. col.All(pred) will return true for an empty collection, so you can leave just `FilteredNotes.Filter = x => SelectedTags.All(t => ((Note)x).Tags.Contains(t));` 
There shouldn't be any performance concerns, so no reason not to prefer a more concise version.